azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/utils/Utils.java

```diff
@@ -50,4 +54,34 @@ public static boolean containsIgnoreCase(List<String> list, String str) {
     public static String getId(Object obj) {
         return Integer.toHexString(System.identityHashCode(obj));
     }
+
+    public static <T> T firstNonNull(T... args) {
+        for (T obj : args) {
+            if (obj != null) {
+                return obj;
+            }
+        }
+        return null;
+    }
+
+    public static <T> T copyProperties(T to, T from) throws IllegalAccessException, InvocationTargetException {
+        Objects.requireNonNull(from);
+        Objects.requireNonNull(to);
+        for (Field field : FieldUtils.getAllFields(from.getClass())) {
+            FieldUtils.writeField(to, field.getName(), FieldUtils.readField(from, field.getName(), true), true);
```

Comment:
For reference filed, we will just copy the reference, is this enough or we also need deep copy for that?

Reply:
No need, just first level of copying